### PR TITLE
issue #46: Add every useful workflow for this repository

### PR DESCRIPTION
**Pull request by ThomasCardin**
The current workflow (build and push docker images to this organisation registry) is working but the workflow it self is missing steps.

Workflow included:
- [X] Lint test
- [X]  Markdown validation
- [X]  ai-cfia repo standard
- [X]  build and push to github container registry (GCR)

### CODE REVIEW

File: .github/workflows/lint-test.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  pull_request:
+    types:
+      - opened
+      - closed
+      - synchronize
+
+jobs:
+  lint-test:
+    uses: ai-cfia/github-workflows/.github/workflows/workflow-lint-test-python.yml@main
+    secrets: inherit
```
Comment (rngadam) [inline .github/workflows/lint-test.yml]: EOF newline

**Review by rngadam — CHANGES_REQUESTED** [review]
also fix markdown check fails

**Comment by ThomasCardin** [comment]
> also fix markdown check fails

it's working now, but failing for another reason

**Comment by rngadam** [comment]
> > also fix markdown check fails
> 
> it's working now, but failing for another reason

Would it be a good idea to turn this into Draft until you've fixed the failing checks?

**Comment by rngadam** [comment]
@ThomasCardin spelling of useful is one l

File: Dockerfile [hunk 2]
```diff
@@ -2,15 +2,20 @@ FROM python:3
...
+ENV PYTHONPATH=${ARG_PYTHONPATH:-/app}
 
 COPY ./requirements.txt .
 
 RUN pip3 install --no-cache-dir -r requirements.txt
 
 COPY . ./
 
-CMD hypercorn -b :$PORT app:app
+CMD hypercorn -b :8080 app:app
```
Comment (rngadam) [inline Dockerfile:21]: PORT should stay as GCR (Google Cloud Run) expects PORT to be specifiable (although it does default to 8080).

https://cloud.google.com/run/docs/container-contract

The ingress container within an instance must listen for requests on 0.0.0.0 on the port to which requests are sent. By default, requests are sent to 8080, but you can configure Cloud Run to send requests to the port of your choice. *Cloud Run injects the PORT environment variable into the ingress container.*

File: README.md [hunk 3]
```diff
@@ -3,45 +3,53 @@
...
+backend to function, you will need to add the missing values:
 
-* **NACHET_AZURE_STORAGE_CONNECTION_STRING**: Connection string to access external storage (Azure Blob Storage).
-* **NACHET_MODEL_ENDPOINT_REST_URL**: Endpoint to communicate with deployed model for inferencing. 
+* **NACHET_AZURE_STORAGE_CONNECTION_STRING**: Connection string to access
+external storage (Azure Blob Storage).
+* **NACHET_MODEL_ENDPOINT_REST_URL**: Endpoint to communicate with deployed
+model for inferencing.
 * **NACHET_MODEL_ENDPOINT_ACCESS_KEY**: Key used when consuming online endpoint.
 * **NACHET_DATA**: Url to access nachet-data repository
```
Comment (rngadam) [inline README.md:49]: URL

File: .mlc_config.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+    "aliveStatusCodes": [999,200,403]
```
Comment (rngadam) [inline .mlc_config.json]: is 999 and 403 required? I thought this was specific to another repo case pointing to LinkedIN URLs.

**Review by rngadam — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 2 above)
Comment (ThomasCardin) [reply Dockerfile:21]: Okay, but we don't use GCR ❓So we don't need to keep the $PORT.

File: README.md (hunk 3 above)
Comment (ThomasCardin) [reply README.md:49]: Not related to this issue.

File: .mlc_config.json (hunk 4 above)
Comment (ThomasCardin) [reply .mlc_config.json]: 200 and 403 is required but 999 is not (you are right, it's used for linkedin url). I will remove it

File: .github/workflows/lint-test.yml (hunk 1 above)
Comment (ThomasCardin) [reply .github/workflows/lint-test.yml]: Done

File: .mlc_config.json (hunk 4 above)
Comment (ThomasCardin) [reply .mlc_config.json]: Done

**Comment by ThomasCardin** [comment]
@rngadam ping. I'm waiting for this pull request before I can proceed with this one: https://github.com/ai-cfia/infra/pull/4. (change de default tag of the docker image inside each kubernetes deployment to `main` see (https://github.com/ai-cfia/infra/pull/4#discussion_r1457940126))

File: Dockerfile (hunk 2 above)
Comment (rngadam) [reply Dockerfile:21]: Why would you not have this though? Seems to be a good practice to have a default value for PORT but allow it to be overriden too.

File: README.md (hunk 3 above)
Comment (rngadam) [reply README.md:49]: we can still fix typos when we open a file. IMO, this is better than having to open PR to fix only typos. We leave things we modify in a better state than we got it.

**Comment by rngadam** [comment]
> @rngadam ping. I'm waiting for this pull request before I can proceed with this one: [ai-cfia/infra#4](https://github.com/ai-cfia/infra/pull/4). (change de default tag of the docker image inside each kubernetes deployment to `main` see ([ai-cfia/infra#4 (comment)](https://github.com/ai-cfia/infra/pull/4#discussion_r1457940126)))

the test is failing (lint-test step failing)

**Comment by ThomasCardin** [comment]
Out of scope of this issue

**Comment by rngadam** [comment]
> Out of scope of this issue

it is not out of scope; tests need to be green in a pull request before it gets merged.

**Comment by MaxenceGui** [comment]
The error was coming from using a Mock object instead of a MagicMock object (to be able to use the `__iter__` ) function. It was solved on

* #53 

Does this PR still need to be pushed to main, or are the workflows being pushed already?

@ThomasCardin @rngadam

**Comment by rngadam** [comment]
> The error was coming from using a Mock object instead of a MagicMock object (to be able to use the `__iter__` ) function. It was solved on
> 
> * [fixes #31: add a docs folder with Nachet inference documentation #53](https://github.com/ai-cfia/nachet-backend/pull/53)
> 
> Does this PR still need to be pushed to main, or are the workflows being pushed already?
> 
> @ThomasCardin @rngadam

@ThomasCardin since this is fixed in main, this pull request should be rebased to get the tests to pass to allow for merging.

**Pull request closed without merging**